[AaronJT1] iP

AaronJT1 opened the pull request:
# Aj Bot Pull Request

## Why use my bot?
* Fast
* Customisable

## Features
1.  Custom task
2. Stores in database

[Visit my repo!](https://github.com/AaronJT1/ip)

> It really brings me joy!


Here is a sample code in my pull request
```java
public void horiLine() {
        System.out.println("---------------------");
}
```


Use `git status` to list all new or modified files that haven't yet been committed.

## Tick your satisfaction with my PR!
- [x] Good
- [ ] Bad

Lets go to the ~~moon~~  **git** ! :rocket:

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

AaronJT1 committed:
Level-0 Increment

Level-0 Increment

AaronJT1 committed:
Increment Level-1

AaronJT1 committed:
Increment Level-2

AaronJT1 committed:
Increment Level-3

AaronJT1 committed:
Increment Level-4

AaronJT1 committed:
A-TextUiTesting

AaronJT1 committed:
Level-5

AaronJT1 committed:
Level-6

AaronJT1 committed:
Increment Level-6

AaronJT1 committed:
Allow changes to be saved and loaded from hard disk

AaronJT1 committed:
Merge branch 'branch-Level-7'

AaronJT1 committed:
Allow deadline dates to be stored as LocalDate objects

AaronJT1 committed:
Merge branch 'branch-Level-8'

AaronJT1 committed:
Improve OOP structure of code

AaronJT1 committed:
Merge branch 'A-MoreOOP'

AaronJT1 committed:
Enable classes to be in packages

AaronJT1 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

AaronJT1 committed:
Set up gradle support for project

AaronJT1 committed:
Add junit test cases

AaronJT1 committed:
Update build.gradle for shadowJar

AaronJT1 committed:
Add first line of codingStandard

AaronJT1 committed:
Add firstline of javaDoc

AaronJT1 committed:
Add find task by keyword feature

AaronJT1 committed:
Change name of AskCommand function to canAskCommand

AaronJT1 committed:
Add coding convention to code

AaronJT1 committed:
Add JavaDocs to all non-private classes/methods

AaronJT1 committed:
Merge branch 'branch-Level-9'

AaronJT1 committed:
Fix merge conflicts

AaronJT1 committed:
Fix merge conflicts with A-CodingStandard

ryantzr1 commented on src/main/java/aj/Aj.java:
```diff
@@ -0,0 +1,179 @@
+     * @throws IndexOutOfRangeException  If user gives an index > taskList size.
+     * @throws IOException               Arose if there is issue updating database.
+     */
+    public boolean canAskCommand(Scanner scanner) throws NoSuchCommandException, EmptyDescriptionException,
+            IndexOutOfRangeException, IOException {
```
Perhaps a more intuitive name here? Like validCommand

GlendaChong replied:
Perhaps abstract this part of the code out into another file, like Parser instead?

GlendaChong replied:
Perhaps abstract this part of the code out into another file, like Parser instead?

ryantzr1 commented on src/main/java/aj/Event.java:
```diff
@@ -0,0 +1,30 @@
+
+    Event(String taskName, boolean isMarked, String fromDt, String toDt) {
+        super(taskName,
+                isMarked);
```
Not sure if this formatting is supposed to be in a single line

ryantzr1 commented on src/main/java/aj/Parser.java:
```diff
@@ -0,0 +1,76 @@
+     */
+    public Task getEventTask(String remaining, boolean isMark) { // takes in command, parse it and return task object
+        String pattern = " (.*) /from (.*?) /to (.*)";
+
```
Perhaps you could add a line of comment to explain what pattern you are trying to match?

ryantzr1 commented on src/main/java/aj/Storage.java:
```diff
@@ -0,0 +1,149 @@
+    public List<Task> initialiseData() { // get data from file, create necessary task objects and returns an array of
+        // task
+//        String localDir = System.getProperty("user.dir");
+//        File file = new File(localDir + "/src/main/data/actualData.txt");
```
Perhaps you could remove commented out code so that it will look cleaner

GlendaChong replied:
I agree with ryantzr1's comment on this. Perhaps you could do that for other parts of the different files as well.

ryantzr1 commented on src/main/java/aj/Task.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    Task(String taskName, boolean completed) {
+        this.taskName = taskName;
```
Perhaps you could add a JavaDoc comment above the constructor to explain its purpose and parameters?

ryantzr1 commented on src/main/java/aj/TaskList.java:
```diff
@@ -0,0 +1,31 @@
+        this.taskList.remove(idx);
+    }
+
+
```
I like how clean this bit of code is 👍

ryantzr1 commented on src/test/java/aj/ParserTest.java:
```diff
@@ -0,0 +1,50 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ParserTest {
+    @Test
```
Perhaps you could add some Javadocs to explain what this test file does?

ryantzr1 commented on src/test/java/aj/UiTest.java:
```diff
@@ -0,0 +1,87 @@
+        TaskList testTaskList = createTestTaskList();
+        Ui testUi = new Ui(testTaskList);
+        String userInput = "todo\n";
+        Scanner scanner = new Scanner(userInput);
```
Perhaps you could refactor the logic of setting up userInput, scanner and command into a separate private method so that you don't have so much code duplication. But good job!

ryantzr1 reviewed (commented):
Overall, I found your code easy to read. There were some small coding standard violations
too. The app looks very well tested. You could also consider splitting up the code more to prevent code duplication. Good job 💯

GlendaChong commented on src/main/java/aj/Aj.java:
```diff
@@ -0,0 +1,179 @@
+    Ui ui;
+
+
+    /**
+     * Request input from user and does the respective actions.
+     *
```
Should it be "Requests" instead?

GlendaChong commented on src/main/java/aj/Ui.java:
```diff
@@ -0,0 +1,118 @@
+public class Ui {
+    TaskList taskList;
+
+    public void horiLine() {
+        System.out.println("---------------------");
+    }
```
Perhaps you could make horiLine as a constant variable instead, and also use a better naming like horizontalLine instead?

GlendaChong reviewed (commented):
Overall, I found your code easy to read for the most part except a few places
where the nesting was too deep. I noted a few minor coding standard violations
too. Some of the classes are getting quite long. Consider splitting into
smaller classes if that makes sense. Great job so far!

AaronJT1 committed:
Add a GUI for the Aj-Bot

AaronJT1 committed:
Merge branch 'branch-level-10'

AaronJT1 committed:
Add assertions to key places in the code

Currently, there is no assertions put in place to document important
assumptions in the code.

Assertions is important to ensure workability of code and to
reveal any potentials bugs in the logic.

Add assertions to to certain places in the code, mainly in Aj.java

Most of the key logic and assumption is in Aj.java, as such most
assertions are done there.

AaronJT1 committed:
Aj.java canAskCommand function : Refactor to more intuitive name 'getBotResponse'

It is important for functions to be named in way readers can understand.

Change function name from 'canAskCommand' to 'getBotResponse'.

'getBotResponse' is an intuitive name as it simply means this function
gets bot's response.

AaronJT1 committed:
Reduce all methods to be within 30 lines of codes

Longer lines of codes causes readers to take longer time to process.

Reduce the following methods to smaller methods :
getBotResponse() in Aj.java, start() in Aj.java.

To reduce length of code, we define new methods to handle the
abstration.

AaronJT1 committed:
Merge pull request #1 from AaronJT1/branch-A-CodeQuality

Revise code to follow code quality standards

AaronJT1 committed:
Merge branch 'master' into branch-A-assertions

AaronJT1 committed:
Merge branch 'master' into branch-A-CodeQuality

AaronJT1 committed:
Merge branch 'master' into branch-A-Assertions

AaronJT1 committed:
Merge pull request #2 from AaronJT1/branch-A-Assertions

Add assertions to key places in the code

AaronJT1 committed:
Merge branch 'master' into branch-A-Assertions

AaronJT1 committed:
Merge branch 'master' into branch-A-CodeQuality

AaronJT1 committed:
Add undo function for certain commands

AaronJT1 committed:
Merge branch 'branch-C-undo'

AaronJT1 committed:
Merge branch 'master' into branch-A-Assertions

AaronJT1 committed:
Merge branch 'branch-A-Assertions'

AaronJT1 committed:
Merge branch 'master' into branch-A-CodeQuality

AaronJT1 committed:
Merge branch 'branch-A-CodeQuality'

AaronJT1 committed:
Add Ui.png into docs folder

AaronJT1 committed:
Update README.md to include GitHub Flavoured MarkDown

AaronJT1 committed:
Update README.md to include GitHub Flavoured MarkDown

AaronJT1 committed:
Update README.md on root directory to that of docs directory

AaronJT1 committed:
Add gradle checkstyle plugin

AaronJT1 committed:
Merge branch 'branch-A-checkstyle'

AaronJT1 committed:
Update README.md to include more features of Aj-Bot

AaronJT1 committed:
Update Aj.java to use relative path for database

AaronJT1 committed:
Fix database.txt issues
